crypto/x509/v3_skid.c

```diff
@@ -105,7 +105,7 @@ static ASN1_OCTET_STRING *s2i_skey_id(X509V3_EXT_METHOD *method,
         return NULL;
     }
 
-    return ossl_x509_pubkey_hash(ctx->subject_req != NULL ?
-                                 ctx->subject_req->req_info.pubkey :
-                                 ctx->subject_cert->cert_info.key);
+    return ossl_x509_pubkey_hash(ctx->subject_cert != NULL ?
+                                 ctx->subject_cert->cert_info.key :
+                                 ctx->subject_req->req_info.pubkey);
```

Comment:
I think this is a good change, but the commit message says that it does not change the semantics "because only either one may be given".  Since this is in the implementation of an `X509V3_EXT_METHOD`, I wasn't entirely sure where to go to confirm that this invariant is actually enforced.  If we can't prove that it's already enforced, I'd suggest toning down the commit message to reflect only expected usage, not an absolute lack of semantic change.

Reply:
Good point - so I toned down the commit message as you suggested.